addons/event_crm/models/event_registration.py

```diff
@@ -190,15 +190,24 @@ def _get_lead_contact_values(self):
 
         :return dict: values used for create / write on a lead
         """
-        valid_partner = related_partner = next(
+        valid_partner = next(
             (reg.partner_id for reg in self if reg.partner_id != self.env.ref('base.public_partner')),
             self.env['res.partner']
         )  # CHECKME: broader than just public partner
 
-        # mono registration mode: keep partner only if email and phone matches, otherwise registration > partner
-        if len(self) == 1:
-            if (related_partner.phone and self.phone and related_partner.phone != self.phone) or \
-                (related_partner.email and self.email and related_partner.email != self.email):
+        # mono registration mode: keep partner only if email and phone matches;
+        # otherwise registration > partner. Note that email format has to be
+        # taken into account in comparison
+        if len(self) == 1 and valid_partner:
+            # compare emails: email_normalized or raw
+            if self.email and valid_partner.email:
+                if valid_partner.email_normalized and tools.email_normalize(self.email) != valid_partner.email_normalized:
+                    valid_partner = self.env['res.partner']
+                elif not valid_partner.email_normalized and valid_partner.email != self.email:
```

Comment:
I seem to remember this is written differently on your recent changes about registration sync, but I could be mistaken.

Reply:
I backported another fix from 16, and there are some changes in the code with fix in 14. So yes there will be conflicts when forward-porting.

Reply:
Oh and in 16, we can call some methods like _phone_format on partner model which does not exist in 14.